src/custom/state/enhancedTransactions/hooks/index.ts

```diff
@@ -20,18 +22,62 @@ export function useTransactionAdder(): TransactionAdder {
     (addTransactionParams: AddTransactionHookParams) => {
       if (!account || !chainId) return
 
-      const { hash, hashType = HashType.ETHEREUM_TX, summary, approval, presign } = addTransactionParams
+      const { hash, hashType, summary, approval, presign, safeTransaction } = addTransactionParams
       if (!hash) {
-        throw Error('No transaction hash found.')
+        throw Error('No transaction hash found')
       }
-      dispatch(addTransaction({ hash, hashType, from: account, chainId, approval, summary, presign }))
+      dispatch(addTransaction({ hash, hashType, from: account, chainId, approval, summary, presign, safeTransaction }))
     },
     [dispatch, chainId, account]
   )
 }
 
-export function useAllPendingHashes(): string[] {
+type TransactionFilter = (tx: EnhancedTransactionDetails) => boolean
+
+/**
+ * Return all transactions details
+ */
+export function useAllTransactionsDetails(filter?: TransactionFilter): EnhancedTransactionDetails[] {
   const transactions = useAllTransactions()
 
-  return useMemo(() => Object.keys(transactions).filter((hash) => !transactions[hash].receipt), [transactions])
+  return useMemo(() => {
+    const transactionsDetails = Object.keys(transactions).map((hash) => transactions[hash])
+
+    return filter ? transactionsDetails.filter(filter) : transactionsDetails
+  }, [transactions, filter])
+}
+
+export type TransactionsByType = Record<HashType, EnhancedTransactionDetails[]>
+
+/**
+ * Return all transactions grouped by type
+ */
+export function useAllTransactionsByType(filter?: TransactionFilter): TransactionsByType {
```

Comment:
you want to keep this cause it's cool code even tho we don't use it 😛 we never do that though ;)

Reply:
Exactly, I think is likely will need it, and it's tree-shaken so doesn't harm to have imo